gap/grahom.gi

```diff
@@ -754,3 +754,233 @@ function(D1, D2)
+      for prev in [1 .. N1] do
+        if defined[prev] then
+          IntersectBlist(conditions[depth + 1][join1[prev][next]],
+                         mask[join2[map[prev]][map[next]]]);
+          IntersectBlist(conditions[depth + 1][meet1[prev][next]],
+                         mask[meet2[map[prev]][map[next]]]);
+        fi;
+      od;
+
+      try_next_value := false;
+      # If the value map[next] breaks a previously defined image of map,
+      # try the next possible value for map[next]
+      for x in [1 .. N1] do
+        if defined[x] and not conditions[depth + 1][x][map[x]] then
+          try_next_value := true;
+          break;
+        fi;
+      od;
+      if not try_next_value then
+        for x in [1 .. N1] do
+          if out1[next][x] then
+            IntersectBlist(conditions[depth + 1][x], out2[value]);
+          else
+            FlipBlist(out2[value]);
+            IntersectBlist(conditions[depth + 1][x], out2[value]);
+            FlipBlist(out2[value]);
+          fi;
+          if in1[next][x] then
+            IntersectBlist(conditions[depth + 1][x], in2[value]);
+          else
+            FlipBlist(in2[value]);
+            IntersectBlist(conditions[depth + 1][x], in2[value]);
+            FlipBlist(in2[value]);
+          fi;
+        od;
+        if Recurse(next, depth + 1, print) then
+          return true;
+        fi;
+      fi;
+      Unbind(conditions[depth + 1]);
+      Unbind(map[next]);
+      not_in_image[value] := true;
+      defined[next]       := false;
+      value := Position(conditions[depth][next], true, value);
+    od;
+    return false;
+  end;
+
+  if Recurse(1, 1, false) then
+    Append(map, [N1 + 1 .. N2]);
+    return p ^ -1 * Transformation(map);
+  fi;
+  return fail;
+end);
+
+InstallMethod(IsLatticeHomomorphism,
+"for a transformation and a pair of digraphs",
+[IsDigraph, IsDigraph, IsTransformation],
+function(L1, L2, map)
+  local N1, N2, x, y, meet1, meet2, join1, join2;
+
+  N1 := DigraphNrVertices(L1);
+  N2 := DigraphNrVertices(L2);
+
+  if LargestMovedPoint(map) > N1 then
+    return false;
+  fi;
+
+  # We compute the join/meet table to avoid having to do this twice if L1 or L2
+  # is mutable
+  join1 := DigraphJoinTable(L1);
+  meet1 := DigraphMeetTable(L1);
+
+  if join1 = fail or meet1 = fail then
+    ErrorNoReturn("the 1st argument (a digraph) must be a lattice digraph");
+  fi;
+
+  join2 := DigraphJoinTable(L2);
+  meet2 := DigraphMeetTable(L2);
+
+  if join2 = fail or meet2 = fail then
+    ErrorNoReturn("the 2nd argument (a digraph) must be a lattice digraph");
+  fi;
+
+  if Maximum(ImageSetOfTransformation(map, N1)) > N2 then
+    return false;
+  fi;
+  # The above checks if the <x ^ map> and <y ^ map> entries of
+  # meet2 and join2 exist
+
+  for x in [1 .. N1] do
+    for y in [1 .. N1] do
+      if meet2[x ^ map, y ^ map] <> meet1[x, y] ^ map
+          or join2[x ^ map, y ^ map] <> join1[x, y] ^ map then
```

Comment:
Again, you might not want to implement this - but do you really have to check this for every pair of elements? I would guess that you can do something with irreducible elements here.

Reply:
That's a good question, I don't immediately know the answer, but will add a comment to the code to indicate that this should be investigated!